src/components/TreeGrid.ts

```diff
@@ -72,6 +75,8 @@ type TreeGridProps<T extends RowContents> = {
   readonly columns: Column<T>[];
   /** The root of the tree to display. Note that the root node is hidden, and only its children are shown. */
   readonly root: T;
+  /** For a pre-initialized tree, all the known parent nodes. */
+  readonly parents?: Parent<T>[];
```

Comment:
Just to confirm, this is effectively: `parent, grandparent, greatgrandparent, etc`? And then they each have their children information stored in `children`?

Would `ancestors` be a better term for this? Just to include the fact that it might be `n` deep?

Reply:
No, this change gets rid of `children`.. it looks like when I updated my branch from main, I lost my changes that removed `children`, I'll have to go back and fix that.

This table is a list of all parents in the tree. For each parent, its direct children are stored in `children`. To get grandchildren you now need to look up each child in the parent list.